src/librustc_mir/interpret/intrinsics.rs

```diff
@@ -216,6 +216,46 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
             "transmute" => {
                 self.copy_op_transmute(args[0], dest)?;
             }
+            "ptr_offset_from" => {
+                let a = self.read_immediate(args[0])?.to_scalar()?.to_ptr()?;
+                let b = self.read_immediate(args[1])?.to_scalar()?.to_ptr()?;
+                if a.alloc_id != b.alloc_id {
+                    throw_ub_format!(
+                        "ptr_offset_from cannot compute offset of pointers into different \
+                        allocations.",
+                    );
+                }
+                let ptr_size = self.tcx.data_layout.pointer_size;
+                let a = Scalar::from_uint(a.offset.bytes(), ptr_size);
```

Comment:
I found the name `a` here a bit confusing -- this is not the same as the `a` it is shadowing. Maybe `a_offset` would be a better name? (Same for `b`, obviously.)